src/OneSignal.php

```diff
@@ -86,6 +87,25 @@ public function sendRequest(RequestInterface $request): array
         return $content;
     }
 
+    public function makeRequest(RequestInterface $request): array
+    {
+        $response = $this->httpClient->sendRequest($request);
+
+        if ($response->getStatusCode() >= 400) {
```

Comment:
probably like, not sure about redirects because clients should follow them :thinking: 
```suggestion
        if ($response->getStatusCode() < 200 || $response->getStatusCode() >= 400) {
```